### PR TITLE
Feature/button component

### PR DESCRIPTION
**Pull request by rakeshkumar-verma**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -189,4 +189,4 @@ Bundle-DocURL:
             <artifactId>slf4j-test</artifactId>
         </dependency>
     </dependencies>
-</project>
+</project>
```
Comment (rishabh-behl) [inline core/pom.xml]: @rakeshkumar-verma : Pl remove this file out of PR.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                                        jcr:primaryType="nt:unstructured"
+                                        sling:resourceType="granite/ui/components/coral/foundation/container">
+                                        <items 
+											jcr:primaryType="nt:unstructured"
+											sling:hideChildren="icon">
+                                            <buttonType
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:orderBefore="text"
+                                                sling:resourceType="granite/ui/components/coral/foundation/form/select"
+                                                fieldLabel="Button Type"
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml]: @rakeshkumar-verma : Button Type will be part of style system . Please add that as policy along with default css class and remove from dialog.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                                    </column>
+                                </items>
+                            </columns>
+                        </items>
+                    </properties>
+                </items>
+            </tabs>
+        </items>
+    </content>
+</jcr:root>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml]: @rakeshkumar-verma : put back the icon deialog field.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/button.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<div class='${properties.buttonType}${properties.disabledButton}'>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/button.html]: @rakeshkumar-verma ; why do we have this div wrapping <button>

**Review by rishabh-behl — CHANGES_REQUESTED** [review]
@rakeshkumar-verma : pl review comments

File: core/pom.xml (hunk 1 above)
Comment (rakeshkumar-verma) [reply core/pom.xml]: done

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml (hunk 2 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml]: review comment incorporated

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml (hunk 3 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/_cq_dialog/.content.xml]: done

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/button.html (hunk 4 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/button.html]: added based previous approach. Now removed this as we are handling button type from style system

**Comment by rishabh-behl** [comment]
LGTM

**Pull request merged**